CMakeLists.txt

```diff
@@ -250,7 +252,9 @@ endif()
 # ---------------------------------------------------------------------------
 
 if (CMAKE_CXX_COMPILER_ID MATCHES "(GNU|Clang)")
-  add_definitions(-fno-math-errno -ffp-contract=fast -fno-trapping-math)
+  add_compile_options($<$<COMPILE_LANGUAGE:CXX>:-fno-math-errno>)
+  add_compile_options($<$<COMPILE_LANGUAGE:CXX>:-ffp-contract=fast>)
+  add_compile_options($<$<COMPILE_LANGUAGE:CXX>:-fno-trapping-math>)
```

Comment:
These looked like compiler options rather than definitions, so I switched `add_definitions` to `add_compile_options`.